src/model.ts

```diff
@@ -298,6 +298,17 @@ export class SpecQueryModel {
           if (!PROPERTY_SUPPORTED_CHANNELS[prop] ||  // all channels support this prop
             PROPERTY_SUPPORTED_CHANNELS[prop][encQ.channel as Channel]) {
             fieldDef[prop] = encQ[prop];
+
+            if (prop === Property.SCALE) {
+              if (isFieldQuery(encQ)) {
+                let field = encQ['field'];
```

Comment:
Please use `.field` and never use string accessor (e.g., `['field']`) to hack around type except inside type guard.  (If you have to hack like this, it's likely that something is wrong.)